### ISSUE TITLE
Editors are extremely slow

### ISSUE THREAD

**Issue by dora-gt** [intellij-rust/intellij-rust#4566]
## Environment

* **IntelliJ Rust plugin version:**
0.2.108.2143-193

* **Rust toolchain version:**
stable-x86_64-apple-darwin (default)
rustc 1.38.0 (625451e37 2019-09-23)

* **IDE name and version:**
CLion 2019.3 EAP
Build #CL-193.4778.13, built on October 23, 2019

* **Operating system:**
CLion EAP User
Expiration date: November 22, 2019
Runtime version: 11.0.4+10-b520.2 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.6
GC: ParNew, ConcurrentMarkSweep
Memory: 4083M
Cores: 8
Registry: run.processes.with.pty=TRUE
Non-Bundled Plugins: Docker, String Manipulation, org.toml.lang, org.rust.lang

## Problem description
When I'm typing, editors are extremely slow.
CLion uses almost 50% of CPU. Really frustrating.

I'm not sure this is one of the causes but I'm developing this project.
https://github.com/interledger-rs/interledger-rs

I'm using one of the Japanese input methods.

## Steps to reproduce
Open a file, type something.

**Comment by dora-gt** [comment]
If I can use CLion with fewer functions and much faster, I'll choose it because typing experience is quite important.

**Comment by prostomarkeloff** [comment]
Have a similar behaviour.

**Comment by splix** [comment]
I can confirm the problem. I was always super slow, but latest nightly releases made it completely unusable.

I also don't need all of the functions, basically a syntax highlighter would cover 80% of my needs, so a possibility to disable unused features would be greatly appreciated.

**Comment by vlad20012** [comment]
@dora-gt I tried to reproduce it on your project and with the same versions of CLion and Rust plugin. But I didn't succeed. I.e. everything works fine for me, the editor is very responsive. BUT I use Linux. I'll try to reproduce it on macOS.

Could you please make and attach here a CPU snapshot? For this, in CLIon click on `Help | Diagnostic | Start CPU Usage Profiling`, then type some text (ideally, no less than 1 minute), then click on `Help | Diagnostic | Stop CPU Usage Profiling.`. Then you should receive a notification with a done snapshot. 

Could you also check that you have enough free memory for CLion, i.e. it's not swapped or something? Maybe it's reasonable to reboot the macOS (sometimes it is very unpredictable IMHO)

**Comment by vlad20012** [comment]
@prostomarkeloff @splix Could you please describe your environment just like in the original issue? It is very important. Otherwise, I have to just ignore your comments.

**Comment by dora-gt** [comment]
@vlad20012 Thank you, here is the CPU usage.

[CL-193.4778.13_dora_gt_31.10.2019_21.40.19.zip](https://github.com/intellij-rust/intellij-rust/files/3793761/CL-193.4778.13_dora_gt_31.10.2019_21.40.19.zip)

I think I have enough memory.

**Comment by prostomarkeloff** [comment]
Thanks, this is my CPU usage snapshot.
I'm using Ubuntu 18.04, 8GB RAM.
[CL-192.7142.39_markeloffdev_31.10.2019_17.17.57.zip](https://github.com/intellij-rust/intellij-rust/files/3794162/CL-192.7142.39_markeloffdev_31.10.2019_17.17.57.zip)

**Comment by dora-gt** [comment]
Is CLion running compilation process or inference process or something during typing? If so, I think it is the cause. We might need some options like: `run XX process after type: [500] msec`

**Comment by prostomarkeloff** [comment]
Any updates?

**Comment by splix** [comment]
For my situation upgrade to `0.2.110.2781-192-nightly` fixed most of the problem. Not _extremely_ slow for me anymore.

Previously I wasn't even able to do _CPU Usage Profiling_ or anything like that. Because with profiling enabled, the IDE was starting to freeze just after typing `pub fn my_fn() {}`. Though it didn't use much of CPU (as I was able to see from System Monitor), it was failing into "Not Responding" status, and only "Force Kill" of the whole IDE app was a solution.

**Comment by dora-gt** [comment]
@splix Where can I get it?

**Comment by dora-gt** [comment]
nvm, I found here:
https://intellij-rust.github.io/install.html

**Comment by dora-gt** [comment]
Hmmm it is still slow. Any updates on this?

**Comment by prostomarkeloff** [comment]
> Hmmm it is still slow. Any updates on this?

same.

**Comment by prof7bit** [comment]
Try switching the JVM runtime away from the bundled JetBrainsJVM to OpenJDK or Oracle. I recently did this with PyCharm IDE and with the other runtime the IDE starts 5 times faster and is responsive again like it used to be. I guess they messed up some compiler switch in the release build of their JVM that makes it at least 5 times slower than any other JVM. I wonder why they need their own JVM at all when there are perfectly working JVMs for free.

**Comment by StarkBotsIndustries** [comment]
And it's 2022 now

**Comment by buehler** [comment]
@prof7bit which JDK version did you use? java 17/18? or a less modern one?

**Comment by AokiAhishatsu** [comment]
> And it's 2022 now
Yep, can confirm: IDEA 2022.1.1 + Rust plugin is still not usable.

Heavy lags during normal typing

**Comment by TobiasDeBruijn** [comment]
Same issue here. Not sure what's causing it. Looking at the system monitor, neither my memory nor CPU are being heavily utilized.

Environment:
- OS: Ubuntu 22.04 LTS
- `uname -r`: 5.15.0-33-generic
- CPU: Ryzen 9 5900X
- Memory: 64GB at 3200Mhz
- IDE: CLion 2022.1.1 build 221.5591.52
- Rust plugin: 0.4.171.4656-221
- Rust compiler: 1.62-beta.2
